### PR TITLE
fix: notification slowness and crashes

### PR DESCRIPTION
**Pull request by Prithpal-Sooriya**
## **Description**

This is a series of fixes added to core libraries. Adding to extension, and soon we will migrate extension to use shared libraries.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/25946?quickstart=1)

## **Related issues**

https://github.com/MetaMask/core/pull/4530
https://github.com/MetaMask/core/pull/4531
https://github.com/MetaMask/core/pull/4532
https://github.com/MetaMask/core/pull/4533
https://github.com/MetaMask/core/pull/4536

https://github.com/MetaMask/metamask-extension/issues/25749

## **Manual testing steps**

1. Create multiple accounts
2. Go to notification settings page
3. Wait for settings to load, and try toggling notifications

Before: some settings would error or not load
After: you should be able to toggle accounts and not see errors.

### **Before**

[](https://github.com/MetaMask/metamask-extension/issues/25749)

### **After**

https://www.loom.com/share/49b582e8c33b4199bdafc994a3f6087f?sid=9f94a885-351b-4fee-84d8-f72c97506e7d

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by matteoscurati — APPROVED** [review]
I found two edge cases regarding state management, but they involve behaviors that a user normally wouldn’t apply and, in any case, they don’t cause issues in managing the extension’s state. We can approve this PR and look into further optimizing this page in the future :)

File: app/scripts/controllers/metamask-notifications/services/feature-announcements.ts [hunk 1]
```diff
@@ -113,10 +113,17 @@ async function fetchFeatureAnnouncementNotifications(): Promise<
...
+    process.env.CONTENTFUL_ACCESS_SPACE_ID &&
+    process.env.CONTENTFUL_ACCESS_TOKEN
+  ) {
+    const rawNotifications = await fetchFeatureAnnouncementNotifications();
+    const notifications = rawNotifications.map((notification) =>
+      processFeatureAnnouncement(notification),
+    );
+
+    return notifications;
+  }
```
Comment (Prithpal-Sooriya) [inline app/scripts/controllers/metamask-notifications/services/feature-announcements.ts:126]: Related shared library change: https://github.com/MetaMask/core/pull/4530

File: app/scripts/controllers/metamask-notifications/services/feature-announcements.ts [hunk 2]
```diff
@@ -26,7 +26,7 @@ export type ContentfulResult = {
 
 async function fetchFromContentful(
   url: string,
-  retries = 3,
+  retries = 1,
```
Comment (Prithpal-Sooriya) [inline app/scripts/controllers/metamask-notifications/services/feature-announcements.ts:29]: https://github.com/MetaMask/core/pull/4531

File: app/scripts/controllers/metamask-notifications/utils/utils.ts [hunk 3]
```diff
@@ -603,7 +603,7 @@ export async function makeApiCall<T>(
   endpoint: string,
   method: 'POST' | 'DELETE',
   body: T,
-  retries = 3,
+  retries = 1,
```
Comment (Prithpal-Sooriya) [inline app/scripts/controllers/metamask-notifications/utils/utils.ts:606]: https://github.com/MetaMask/core/pull/4531

File: app/scripts/controllers/authentication/authentication-controller.ts [hunk 4]
```diff
@@ -299,28 +294,48 @@ export default class AuthenticationController extends BaseController<
...
       createSnapPublicKeyRequest(),
-    ) as Promise<string>;
+    )) as string;
+
+    this.#_snapPublicKeyCache = result;
+
+    return result;
   }
 
+  #_snapSignMessageCache: Record<`metamask:${string}`, string> = {};
```
Comment (Prithpal-Sooriya) [inline app/scripts/controllers/authentication/authentication-controller.ts:319]: https://github.com/MetaMask/core/pull/4532

File: app/scripts/controllers/authentication/authentication-controller.ts [hunk 5]
```diff
@@ -271,8 +268,6 @@ export default class AuthenticationController extends BaseController<
       };
     } catch (e) {
       console.error('Failed to authenticate', e);
-      // Disable Profile Syncing
-      this.messagingSystem.call('UserStorageController:disableProfileSyncing');
```
Comment (Prithpal-Sooriya) [inline app/scripts/controllers/authentication/authentication-controller.ts:275]: https://github.com/MetaMask/core/pull/4533

File: app/scripts/controllers/metamask-notifications/metamask-notifications.ts [hunk 6]
```diff
@@ -278,22 +278,34 @@ export class MetamaskNotificationsController extends BaseController<
 
   #pushNotifications = {
     enablePushNotifications: async (UUIDs: string[]) => {
-      return await this.messagingSystem.call(
-        'PushPlatformNotificationsController:enablePushNotifications',
-        UUIDs,
-      );
+      try {
```
Comment (Prithpal-Sooriya) [inline app/scripts/controllers/metamask-notifications/metamask-notifications.ts:281]: https://github.com/MetaMask/core/pull/4536

**Review by mathieuartu — APPROVED** [review]

**Pull request merged**